### ISSUE TITLE
Graph.deme should return the newly created Deme

### ISSUE THREAD

**Issue by jeromekelleher** [grahamgower/demes#111]
It's often handy to have a reference to the thing you've created, and it does no harm when you don't use it.

**Comment by grahamgower** [comment]
I'm not opposed to this, but the deme is available using `DemeGraph.__getitem__()`.
```
>>> import demes
>>> g = demes.DemeGraph(description="test", time_units="generations")
>>> g.deme("A", initial_size=10)
>>> g["A"]
Deme(id='A', description=None, ancestors=None, proportions=None, epochs=[Epoch(start_time=inf, end_time=0, initial_size=10, final_size=10, size_function='constant', selfing_rate=None, cloning_rate=None)], selfing_rate=None, cloning_rate=None)
```
We also have a `DemeGraph.__contains__()` that facilitates checking if a deme is in the graph.
```
>>> "A" in g
True
```

If we *did* return the newly created `Deme`, should we also have `migration()` return the `Migration`, and `pulse()` return the `Pulse`?

**Comment by jeromekelleher** [comment]
> If we did return the newly created Deme, should we also have migration() return the Migration, and pulse() return the Pulse?

I think it's a good idea for all these methods to return the newly created item. It's a common pattern to want to do something with the thing you've created, and you shouldn't have to type your string ID twice to do this. 

It's not like we want the return value for anything else anway, is it?

**Comment by grahamgower** [comment]
Ok, makes sense. It's an easy change and doesn't compilcate anything, so I'm happy for it to go in.

**Comment by noscode** [comment]
Should `symmetric_migration()` return the list of created migrations?

**Comment by grahamgower** [comment]
Yeah, that seems reasonable.

**Comment by apragsdale** [comment]
Closed by #137.
